dotnet/src/webdriver/VirtualAuth/IHasVirtualAuthenticator.cs

```diff
@@ -31,18 +34,23 @@ public interface IHasVirtualAuthenticator
         /// </summary>
         /// <param name="options">The VirtualAuthenticatorOptions to use in creating the authenticator.</param>
         /// <returns>The ID of the added virtual authenticator.</returns>
+        /// <exception cref="ArgumentNullException">If <paramref name="options"/> is <see langword="null"/>.</exception>
```

Comment:
Do you use Gen AI to write it?

Reply:
No, that sounds like more work than copy-paste 😁 

Visual Studio has some nice fixers and tab-complete functionality for XMLDoc, better than one may expect.

Reply:
OK, let's continue. In my mind interface is a declaration. It cannot declare what type of exception it **might** throw.

Reply:
In my mind, it is the responsibility of an implementation to match the behaviour which the interface documents. This should apply for exceptions too.

Reply:
I am with you, and not at the same time. An implementation may accept `null` as an argument, why not, but the interface dictates to throw if `null`.

If you can provide a link how dotnet team does it, it will help to decide.

Reply:
Looking at the docs for a [random common interface method](https://learn.microsoft.com/en-us/dotnet/api/system.collections.ienumerator.reset?view=net-8.0#exceptions), I see the exceptions are documented.

I think it’s all a question of the contract. Is an interface implementation expected to throw the given exception? In this case, I think yes, and if another implementation does not throw this specific exception in this scenario, that would be a violation of the interface contract. The exception *should* be thrown on `null`.